src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.netcore.cs

```diff
@@ -13019,6 +13019,105 @@ private static Vector512<T> IsNegative<T>(Vector512<T> vector)
             return Vector512.LessThan(vector, Vector512<T>.Zero);
         }
 
+        /// <summary>poly = C0 + C1*r^3 + C2*r^5 + C3*r^7 + C4*r^9</summary>
+        private static Vector128<T> PolyEvalOdd9<T>(Vector128<T> r, Vector128<T> c0, Vector128<T> c1, Vector128<T> c2, Vector128<T> c3, Vector128<T> c4)
```

Comment:
We should not introduce these helpers here and rather should inline the logic directly into the algorithm, as has been done for the other ports.

In most cases these are macros in the C implementation, and thus not actual separate function calls. Having them be function calls, especially with this many large parameters, is likely going to hit several known JIT pessimizations around large structs and forward substitution. It may even not be inlined in some scenarios.

Reply:
It does mean we don't get the nicety of having a reusable helper, but given these are core functions, I think the trade-off is well worth it.

Reply:
Even if we AggressiveInlining on them?  These are going to be reused in multiple operators; it's unfortunate to duplicate them.

Reply:
Yes. Because they are large structs, some platforms will force them to be passed by shadow copy.

In order to inline this, it requires creating `n` new locals with copies that the JIT then has to remove if it realizes they aren't actually mutated. However, the JIT has limits around the number of locals it tracks, so introducing too many is bad.

There is some effort to do forward substitution to avoid introducing locals unnecessarily, but this is limited due to the analysis being pretty expensive. In cases like this, we can hit a pessimization where only the last parameter can be substituted in, which can lead to downstream deoptimizations (such as https://github.com/dotnet/runtime/issues/12308)

This is really the "core" of the algorithm and the most important part, so I think the trade-off ends up being well worth it.

Reply:
I don't love it. But I did it.